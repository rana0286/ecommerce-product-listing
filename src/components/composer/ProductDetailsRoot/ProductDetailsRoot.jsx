import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { ProductListHeader, ProductsDetailsList } from '../../module';

class ProductDetailsRoot extends PureComponent {
  render() {
    return (<div>
      <ProductListHeader
        headingLabel={this.props.headingLabel}
        onSelect={this.props.onSelect}
        productSizeOptions={this.props.productSizeOptions}
      />
      <ProductsDetailsList productsList={this.props.productsList}/>
    </div>);
  }
}

ProductDetailsRoot.propTypes = {
  headingLabel: PropTypes.string.isRequired,
  onSelect: PropTypes.func.isRequired,
  productSizeOptions: PropTypes.arrayOf(PropTypes.object).isRequired,
  productsList: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default ProductDetailsRoot;
