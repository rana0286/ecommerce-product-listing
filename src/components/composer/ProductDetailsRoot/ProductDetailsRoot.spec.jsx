import React from 'react';
import ProductDetailsRoot from './ProductDetailsRoot';
import renderer from 'react-test-renderer';

describe('src/components/composer/ProductDetailsRoot', () => {
  const product = {
    "size": [
      "XS",
      "S",
      "L",
      "XL"
    ],
    "productName": "Striped shirt",
    "productImage": "product-1.jpg",
    "price": "$18.88",
    "isExclusive": false,
    "isSale": true,
    "index": 0
  };
  it('ProductDetailsRoot component should render', () => {
    const tree = renderer.
      create(<ProductDetailsRoot
        headingLabel="Test"
        onSelect={jest.fn()}
        productSizeOptions={[{value: 'Test', label: 'Test'}]}
        productsList={[product]}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
