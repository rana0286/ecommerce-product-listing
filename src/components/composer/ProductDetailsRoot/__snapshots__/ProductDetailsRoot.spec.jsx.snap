// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/composer/ProductDetailsRoot ProductDetailsRoot component should render 1`] = `
<div>
  <div
    className="product-list-header"
  >
    <div
      className="product-list-header__heading"
    >
      <h2
        className="heading heading--h2"
      >
        Test
      </h2>
    </div>
    <div
      className="product-list-header__filter"
    >
      <div
        className="select-box "
      >
        <label
          className="select-box__label"
          htmlFor="select-box"
        >
          
        </label>
        <select
          className="select-box__select"
          id="select-box"
          onChange={[Function]}
          onSelect={[MockFunction]}
          selected=""
          value=""
        >
          <option
            value="Test"
          >
            Test
          </option>
        </select>
      </div>
    </div>
  </div>
  <ul
    className="product-list"
  >
    <li
      className="card"
    >
      <img
        alt="Striped shirt"
        className="image"
        src="./assets/img/products/product-1.jpg"
      />
      <div
        className="product"
      >
        <div
          className="tag"
        >
          <div
            className="tag--sale"
          >
            Sale
          </div>
        </div>
        <div
          className="description"
        >
          <div
            className="description__name"
          >
            Striped shirt
          </div>
          <div
            className="description__price"
          >
            $18.88
          </div>
        </div>
      </div>
    </li>
  </ul>
</div>
`;
